sdk/core/core/internal/az_precondition_internal.h

```diff
@@ -60,27 +60,35 @@ az_precondition_failed_fn az_precondition_failed_get_callback();
 #define AZ_PRECONDITION_NOT_NULL(arg) AZ_PRECONDITION((arg != NULL))
 #define AZ_PRECONDITION_IS_NULL(arg) AZ_PRECONDITION((arg == NULL))
 
-
-AZ_NODISCARD AZ_INLINE bool az_span_is_valid(az_span span, int32_t min_length, bool null_is_valid)
+AZ_NODISCARD AZ_INLINE bool az_span_is_valid(az_span span, int32_t min_size, bool null_is_valid)
 {
-  int32_t span_length = az_span_length(span);
-  int32_t span_capacity = az_span_capacity(span);
-  /* Valid Span is:
-      If the length is greater than or equal to a user defined minimum value AND one of the
+  uint8_t* ptr = az_span_ptr(span);
+  int32_t const span_size = az_span_size(span);
+
+  bool result = false;
+
+  /* Span is valid if:
+     The size is greater than or equal to a user defined minimum value AND one of the
      following:
-        - If null_is_valid is true and the pointer in the span is null, the length and capacity must
-     also be 0. In the case of the pointer not being NULL, two conditions must be met:
-        - The length is greater than or equal to zero and the capacity is greater than or equal to
-          the length.
+        - If null_is_valid is true and the pointer in the span is null, the size must also be 0.
+        - In the case of the pointer not being NULL, the size is greater than or equal to zero.
   */
-  return (
-      (((null_is_valid && (az_span_ptr(span) == NULL) && (span_length == 0) && (span_capacity == 0))
-        || (az_span_ptr(span) != NULL && (span_length >= 0) && (span_capacity >= span_length)))
-       && min_length <= span_length));
+  if (null_is_valid)
+  {
+    result = ptr == NULL ? span_size == 0 : span_size >= 0;
+  }
+  else
+  {
+    result = ptr != NULL && span_size >= 0;
+  }
+
+  result = result && min_size <= span_size;
```

Comment:
You could combine this line and the below line together:
return result && min_size <= span_size;